log4j-core/src/main/java/org/apache/logging/log4j/core/layout/GelfLayout.java

```diff
@@ -598,12 +636,22 @@ private StringBuilder toText(final LogEvent event, final StringBuilder builder,
         if (event.getThrown() != null || layout != null) {
             builder.append("\"full_message\":\"");
             if (layout != null) {
-                final StringBuilder messageBuffer = getMessageStringBuilder();
-                layout.serialize(event, messageBuffer);
-                JsonUtils.quoteAsString(messageBuffer, builder);
+                final StringBuilder messageBuffer = acquireStringBuilder();
```

Comment:
Interesting little use case where you can now abuse the recursion support in `ThreadLocalRecyclerFactory` to get multiple recyclable objects. Already works as expected with the `QueueingRecyclerFactory` version.